ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,216 @@
+        firing_alerts = []
+        for alert in alert_list:
+            if (
+                alert["labels"]["alertname"] == alert_name
+                and alert["state"] == "firing"
+            ):
+                firing_alerts.append(alert)
+
+        if not firing_alerts:
+            raise NoActiveAlertsFoundError(
+                f"No active alert found in firing state for alert: {alert_name}"
+            )
+
+        return firing_alerts
+
+    def wait_for_firing_alert_sampler(self, alert_name, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for an alert and raises timeout exception
+
+        Args:
+             alert(str): alert name
+             timeout(int): wait time, default is 10 mins
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.get_firing_alerts,
+            alert_name=alert_name,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample:
+                    LOGGER.info(f"Found alert: {alert_name} in firing state.")
+                    return
+
+        except TimeoutExpiredError:
+            LOGGER.error(f"{alert_name} is not fired")
+            raise
+
+    def get_scrape_interval(self):
+        """
+        get prometheus scrap interval
+
+        Returns:
+             int : scrape time interval
+        """
+
+        response = self._get_response(query=f"{self.api_v1}/targets")
+        result = response["data"]["activeTargets"]
+        for item in result:
+            if item and item["labels"]["job"] == "prometheus-k8s":
+                scrape_interval = item["scrapeInterval"]
+                scrape_interval = int((re.match(r"\d+", scrape_interval)).group())
+                break
+        return scrape_interval
+
+    def query_sampler(self, query, timeout=TIMEOUT_2MIN):
+        """
+        Sample output for query function
+
+        Args:
+             query(str): prometheus query string
+             wait_timeout(int): default is 10 mins
+
+        Return:
+             list: return the query result
+
+        Raise:
+             TimeoutExpiredError: if query response doesn't return success
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.query,
+            query=query,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                result = sample["data"]["result"]
+                if result and sample["status"] == "success":
+                    return result
+        except TimeoutExpiredError:
+            LOGGER.error(
+                f"Failed to get successful status after executing query '{query}'"
+            )
+            raise
+
+    @property
```

Comment:
This is a property but query is not. We might want consistency.

Reply:
i'll use it as function